lib/mllp/client.ex

```diff
@@ -0,0 +1,444 @@
+defmodule MLLP.ClientContract do
+  @type error_type :: :connect_failure | :send_error | :recv_error
+  @type error_reason :: :closed | :timeout | :no_socket | :inet.posix()
+
+  @type sender_error :: MLLP.Client.Error.t()
+
+  @type options :: [
```

Comment:
Some  of the names of these options could perhaps be better, but not a big deal either.